test/integration/redirect_integration_test.cc

```diff
@@ -145,7 +151,7 @@ TEST_P(RedirectIntegrationTest, BasicInternalRedirect) {
   EXPECT_EQ("200", response->headers().getStatusValue());
   EXPECT_EQ(1, test_server_->counter("cluster.cluster_0.upstream_internal_redirect_succeeded_total")
                    ->value());
-  EXPECT_THAT(waitForAccessLog(access_log_name_), HasSubstr("internal_redirect"));
+  EXPECT_THAT(waitForAccessLog(access_log_name_), AllOf(HasSubstr("internal_redirect"), HasSubstr("302")));
```

Comment:
Please verify that the final 200 response is also present.

Reply:
Done. (A lot of them actually return 302 as the final response.)